test/fixtures/gulpfiles/gulpfile-parallel-failure.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var gulp = require('gulp');
+
+function noop(cb) {
+  cb();
+}
+
+function errorFunction() {
+  throw new Error('Error!');
```

Comment:
you're not completing this task, `cb(new Error('Error!'))`.

Reply:
This should be fine, the domains will catch it.

Reply:
That's what I thought too. Our documentation makes it clear synchronous tasks aren't supported, and throwing an error is technically synchronous.

Reply:
I changed the test. But the effect is the same. An exception thrown synchronously is already catched and considered as a failure completion.